tsk3.c

```diff
@@ -484,7 +484,7 @@ static uint64_t File_read_random(File self, TSK_OFF_T offset,
     RaiseError(EInvalidParameter, "id parameter is invalid.");
     return 0;
   };
-  if(id > 0) {
+  if(id >= 0) {
```

Comment:
Doesn't this change means the else never returns successfully, since it will hit line 493 an you would need to determine the id of the default (nameless) data attribute?

From http://www.sleuthkit.org/sleuthkit/docs/api-docs/4.5/group__fslib.html#ga603ea8f80f1ffbf6be888aeaa508fc81

```
a_id | The id of attribute to load (use 0 and set a_flags if you do not care)
```

http://www.sleuthkit.org/sleuthkit/docs/api-docs/4.5/group__fslib.html#ga54806cf281c7879dcaf7042ff267f447

Reply:
The else is taken when you do not specify an ID, as a default ID of -1 is specified in the python part (don't remember where, maybe in the generated `pytsk3.c`). This is the case when you specify `offset` and `length` only (for instance https://github.com/py4n6/pytsk/blob/886a4bab2979f780e26680b32aec8a50350fe2d2/examples/icat.py#L61)

If you do not care about ID, you can set 0 (or whatever ID, as it's not checked at all in this case) and set the `TSK_FS_FILE_READ_FLAG_NOID` flag (https://github.com/sleuthkit/sleuthkit/blob/9e4bc4b5c0d34dcb266191558337be1e199af3da/tsk/fs/tsk_fs.h#L696)

In `tsk_fs_file_read_type` (https://github.com/sleuthkit/sleuthkit/blob/9e4bc4b5c0d34dcb266191558337be1e199af3da/tsk/fs/fs_file.c#L474), the `fs_attr` value is retrieved using the `type` (and `id` if the `TSK_FS_FILE_READ_FLAG_NOID` flag is not specified)
https://github.com/sleuthkit/sleuthkit/blob/9e4bc4b5c0d34dcb266191558337be1e199af3da/tsk/fs/fs_file.c#L499  
Adn then the content of the attribute is read with `tsk_fs_attr_read`.

This is also what happens in `tsk_fs_file_read` where the default attribute is retrieved with `tsk_fs_file_attr_get` and then also read with `tsk_fs_attr_read`


To sum things up,  if we didn't need the `id` parameter, we should set the `TSK_FS_FILE_READ_FLAG_NOID` flag to retrieve the content of a an attribute (specified by `type`) without caring about its ID.
But as we could need to retrieve an `id` with value 0, this change is needed and it should be the caller responsibility to set the `TSK_FS_FILE_READ_FLAG_NOID` flag if he wants data from a specific `type` but without specifying the ID. (as is done in libtsk actually)

Reply:
I'm moving this to a separate change.